EpsilonLights/Src/Lights.c

```diff
@@ -154,6 +151,46 @@ void blinkSignalLightsTask(void const* arg)
     }
 }
 
+void updateStrobeLight(void const* arg)
+{
+    uint32_t prevWakeTime = osKernelSysTick();
+    // Store inputs values
+    // NOTE: All Lights Out pins are active low
+
+    // If blinkerTimer is within (0 - BLINKER_FREQ), turn blinkers on
+    // If blinkerTimer is within (BLINKER_FREQ - BLINKER_FREQ*2), keep blinkers off
+    // If blinkerTimer is greater than (BLINKER_FREQ*2) reset blinkerTimer to 0
+    uint32_t blinkerTimer = 0;
+    char strobeLight;
+
+    for (;;)
+    {
+        osDelayUntil(&prevWakeTime, LIGHTS_UPDATE_FREQ);
+        strobeLight = (auxBmsInputs[1] >> 0) & STROBE_FAULT_MASK;
+
+        /*Update BMS Strobe*/
+        if (strobeLight && (blinkerTimer <= BLINKER_FREQ))
+        {
+            HAL_GPIO_WritePin(ESTROBE_GPIO_Port, ESTROBE_Pin, LIGHT_ON);
+        }
+        else
+        {
+            HAL_GPIO_WritePin(ESTROBE_GPIO_Port, ESTROBE_Pin, LIGHT_OFF);
+        }
+
+        // Update blinker timer
+        if ((blinkerTimer > BLINKER_FREQ * 2) || !strobeLight)
+        {
+            // If blinkerTimer is greater than (BLINKER_FREQ*2) reset blinkerTimer to 0
+            blinkerTimer = 0;
+        }
+        else
```

Comment:
There should be one more `else if` where if `!strobeLight`, `blinkerTimer` gets set to 0. That way, the strobe light will come on as soon as there is a trip, and doesn't have to wait for the next cycle.

Reply:
Im not sure i understand. Doesn't blinker timer already get set to zero if !strobeLight and the strobe light comes on immediately because blinkerTimer will start off as 0?

Reply:
Ah yup you're right, I read it wrong.